### PR TITLE
number of groups

### LINKED ISSUE #3352: Add a column to the price list grid to show number of groups it modifies
It would also be nice to note how many groups are currently using a price list.  Similar to how the patient groups, debtor groups, and others link the lists from their main page, it would be nice to have "{{ x }} groups using this list" as a column.

### PR DESCRIPTION
**Pull request by jeremielodi**
closes #3352

### CODE REVIEW

File: client/src/modules/prices/prices.js [hunk 1]
```diff
@@ -48,6 +48,12 @@ function PriceListController(
       headerCellFilter : 'translate',
       cellTemplate : `/modules/prices/templates/itemsNumber.cell.html`,
     },
+    {
+      field : 'subcribedGroupsNumber',
+      displayName : 'PATIENT_GROUP.PATIENT_GROUP',
+      headerCellFilter : 'translate',
+      cellTemplate : `/modules/prices/templates/subscribedGroups.cell.html`,
```
Comment (jniles) [inline client/src/modules/prices/prices.js:55]: Why did you need to make a cell for this?

File: server/controllers/finance/priceList.js [hunk 2]
```diff
@@ -36,9 +36,16 @@ function lookup(req) {
 
   if (req.query.detailed === '1') {
     sql = `
-      SELECT BUID(pl.uuid) as uuid, pl.label, pl.created_at, pl.description, COUNT(pl_it.uuid) as itemsNumber
+      SELECT BUID(pl.uuid) as uuid, pl.label, pl.created_at, pl.description,
+        COUNT(pl_it.uuid) as itemsNumber, groups.nbr as subcribedGroupsNumber
       FROM price_list pl
       LEFT JOIN price_list_item pl_it ON pl_it.price_list_uuid = pl.uuid
+      LEFT JOIN (
```
Comment (jniles) [inline server/controllers/finance/priceList.js]: Why does this need to be a subquery?  Can't we do this in the same format as the `itemsNumber`?

File: client/src/modules/patients/groups/groups.html [hunk 3]
```diff
@@ -26,18 +26,20 @@
           <div class="panel-heading">
             <span translate> PATIENT_GROUP.ALL_PATIENT_GROUP </span>
           </div>
-          <table class="table table-condensed">
+          <table class="table table-condensed table-bordered">
             <thead>
               <tr>
                 <th translate> TABLE.COLUMNS.NAME </th>
                 <th translate> TABLE.COLUMNS.PRICE_LIST </th>
+                <th translate> FORM.LABELS.PATIENT</th>
```
Comment (jniles) [inline client/src/modules/patients/groups/groups.html]: Looks like this should be `PATIENTS`.

File: server/controllers/medical/patientGroups.js [hunk 4]
```diff
@@ -30,8 +30,15 @@ function list(req, res, next) {
   const filters = new FilterParser(req.query, { tableAlias : 'pg' });
   const sql = `
     SELECT BUID(pg.uuid) as uuid, pg.name, BUID(pg.price_list_uuid) AS price_list_uuid,
-      pg.note, pg.created_at, pl.label AS priceListLabel, pl.description
-    FROM patient_group AS pg LEFT JOIN price_list AS pl ON pg.price_list_uuid = pl.uuid
+      pg.note, pg.created_at, pl.label AS priceListLabel, pl.description,
```
Comment (jniles) [inline server/controllers/medical/patientGroups.js:33]: Is this needed for the Price List group count...?

**Review by jniles — COMMENTED** [review]
It looks like you've combined two pull requests into one.  What needed to change in patient groups?

File: client/src/modules/prices/prices.js (hunk 1 above)
Comment (jeremielodi) [reply client/src/modules/prices/prices.js:55]: Hum..., I was trying to add this subscribers number where possible to end with this problem

File: server/controllers/finance/priceList.js (hunk 2 above)
Comment (jeremielodi) [reply server/controllers/finance/priceList.js]: There is an important ```GROUP BY``` in the subquery

File: client/src/modules/prices/prices.js (hunk 1 above)
Comment (jniles) [reply client/src/modules/prices/prices.js:55]: But do you need a template?  What happens if you remove the cell template?

File: server/controllers/medical/patientGroups.js (hunk 4 above)
Comment (jeremielodi) [reply server/controllers/medical/patientGroups.js:33]: It's used in the patient group grid, we also need to know how many patients belong to a group

File: server/controllers/finance/priceList.js (hunk 2 above)
Comment (jniles) [reply server/controllers/finance/priceList.js]: Try
```sql
LEFT JOIN patient_group ON patient_group.price_list_uuid = pl.uuid
```

And then put `COUNT(patient_group.uuid)  as subcribedGroupsNumber` in the SELECT part.  I think ti should behave the same, but save quite a bit of work.  JOINs are typically more standard than writing subqueries.

**Review by jniles — APPROVED** [review]
LGTM.

bors r+
